cpp/src/sampling/detail/sampling_utils_impl.cuh

```diff
@@ -78,26 +78,63 @@ count_and_remove_duplicates(raft::handle_t const& handle,
     std::move(result_src), std::move(result_dst), std::move(result_wgt), std::move(result_count));
 }
 
-template <typename vertex_t, typename weight_t>
+template <typename vertex_t>
 struct return_all_edges_e_op {
-  template <typename W = weight_t>
-  __device__
-    std::enable_if_t<std::is_same_v<W, void>, thrust::optional<thrust::tuple<vertex_t, vertex_t>>>
-      __device__ operator()(
-        vertex_t src, vertex_t dst, thrust::nullopt_t, thrust::nullopt_t, thrust::nullopt_t)
+  template <typename EdgeProperties>
+  auto __host__ __device__ operator()(vertex_t src,
+                                      vertex_t dst,
+                                      thrust::nullopt_t,
+                                      thrust::nullopt_t,
+                                      EdgeProperties edge_properties)
   {
-    return thrust::make_optional(thrust::make_tuple(src, dst));
+    if constexpr (std::is_same_v<EdgeProperties, thrust::nullopt_t>) {
+      return thrust::make_optional(thrust::make_tuple(src, dst));
+    } else if constexpr (std::is_arithmetic<EdgeProperties>::value) {
+      return thrust::make_optional(thrust::make_tuple(src, dst, edge_properties));
+    } else if constexpr (cugraph::is_thrust_tuple_of_arithmetic<EdgeProperties>::value &&
+                         (thrust::tuple_size<EdgeProperties>::value == 2)) {
+      return thrust::make_optional(thrust::make_tuple(
+        src, dst, thrust::get<0>(edge_properties), thrust::get<1>(edge_properties)));
+    } else if constexpr (cugraph::is_thrust_tuple_of_arithmetic<EdgeProperties>::value &&
+                         (thrust::tuple_size<EdgeProperties>::value == 3)) {
+      return thrust::make_optional(thrust::make_tuple(src,
```

Comment:
I wonder, what if we have EdgeProperties are tuple of size 4 or more?

Reply:
Use thrust_tuple_cat & to_thrust_tuple.

https://github.com/rapidsai/cugraph/blob/branch-23.02/cpp/include/cugraph/utilities/thrust_tuple_utils.hpp#L215
https://github.com/rapidsai/cugraph/blob/branch-23.02/cpp/include/cugraph/utilities/thrust_tuple_utils.hpp#L201

Something like
```
return thrust::make_optional(thrust_tuple_cat(thrust::make_tuple(src), thrust::make_tuple(dst), to_thrust_tuple(edge_properties)));
```

Reply:
Added a FIXME about using `thrust_tuple_cat`